pkg/services/accesscontrol/accesscontrol.go

```diff
@@ -16,3 +17,41 @@ type AccessControl interface {
 	// Middleware checks if service disabled or not to switch to fallback authorization.
 	IsDisabled() bool
 }
+
+func HasAccess(ac AccessControl, c *models.ReqContext) func(fallback func(*models.ReqContext) bool, permission string, scopes ...string) bool {
+	return func(fallback func(*models.ReqContext) bool, permission string, scopes ...string) bool {
+		if ac.IsDisabled() {
+			return fallback(c)
+		}
+
+		hasAccess, err := ac.Evaluate(c.Req.Context(), c.SignedInUser, permission, scopes...)
+		if err != nil {
+			c.Logger.Error("Error from access control system", "error", err)
+			return false
+		}
+
+		return hasAccess
+	}
+}
+
+var ReqGrafanaAdmin = func(c *models.ReqContext) bool {
+	return c.IsGrafanaAdmin
+}
+
+var ReqOrgAdmin = func(c *models.ReqContext) bool {
+	return c.OrgRole == models.ROLE_ADMIN
+}
+
+var ReqOrgAdminOrEditor = func(c *models.ReqContext) bool {
+	return c.OrgRole == models.ROLE_ADMIN || c.OrgRole == models.ROLE_EDITOR
+}
+
+var ReqOrgAdminOrEditorsCanAdmin = func(cfg *setting.Cfg) func(*models.ReqContext) bool {
+	return func(c *models.ReqContext) bool {
+		return c.OrgRole == models.ROLE_ADMIN || (cfg.EditorsCanAdmin && c.OrgRole == models.ROLE_EDITOR)
+	}
+}
+
+var ReqSignedIn = func(c *models.ReqContext) bool {
+	return c.IsSignedIn
+}
```

Comment:
Could we add these when we need them? I think there are some things we want to adopt from https://github.com/grafana/grafana/blob/610999cfa24896d97f1991deae43c6d9f0e1f172/pkg/middleware/auth.go#L97-L130